libgalois/include/katana/PropertyGraph.h

```diff
@@ -403,8 +410,14 @@ class KATANA_EXPORT PropertyGraph {
 
   const GraphTopology& topology() const { return topology_; }
 
+  /// Add Node properties that do not exist in the current graph
   Result<void> AddNodeProperties(const std::shared_ptr<arrow::Table>& props);
+  /// Add Edge properties that do not exist in the current graph
   Result<void> AddEdgeProperties(const std::shared_ptr<arrow::Table>& props);
+  /// If column name exists, replace it, otherwise insert it
+  Result<void> UpsertNodeProperties(const std::shared_ptr<arrow::Table>& props);
+  /// If column name exists, replace it, otherwise insert it
+  Result<void> UpsertEdgeProperties(const std::shared_ptr<arrow::Table>& props);
```

Comment:
Suggestion (feel free to ignore): rather than adding totally separate interfaces I think this functionality would make sense as a flag instead. Something like:
`Result<void> AddNodeProperties(const std::shared_ptr<arrow::Table>& props, bool overwrite = false);`

Reply:
Well, I generally prefer descriptive names to Boolean flags.  But I think maybe your underlying message is that we don't want to needlessly proliferate functions.  I could imagine getting rid of `AddProperties`, but I think that function has nice error checking for clients.

Reply:
Yeah I though it might reduce the duplication a bit, but you're right that the public interface isn't really the place to do that.

Reply:
I tend to agree with @witchel about names over flags.

re: Upsert + Add versus Upsert only, I can see a reason for having both; though that would suggest we are might want "replace" (update an existing property) eventually.